### PR TITLE
Consolidate the channels + channel-labels

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
Further work on https://github.com/web-platform-tests/wpt.fyi/issues/535 

Changes the TestRunQuery to expand channel-labels onto the products during the building phase, and collapse them off of the products when producing the query-string.

## Review Information
- Visit /flags, enable queryBuilder
- Load /results, open the query builder with the pencil icon
- Observe that the Channel is `Stable` for all products
- Submit the query
- Observe that the query is ?aligned&label=stable, not ?product=chrome[stable]&product=...

### CODE REVIEW

**Comment by foolip** [comment]
This is great! I could quite easily use it get a view of Chrome+Edge+Firefox stable and Safari experimental, which was a URL I was trying to make manually just yesterday.

I also think I've found a bug, so before I review here are steps to repro:
1. Enable /flags and start at https://shared-channel-dot-wptdashboard-staging.appspot.com/results/
1. Click the pen
1. Change all the browser channels to "Experimental", uncheck "Align runs" and submit
1. You are at https://shared-channel-dot-wptdashboard-staging.appspot.com/results/?label=experimental
1. Change all the browser channels back to "Stable" and submit
1. You are *still* at ?label=experimental. Even toggling "Align runs", changing back and forth, as long as you pick "Stable" for all, you'll be back at ?label=experimental.

**Comment by foolip** [comment]
Oh, it also looks like "Add product" is broken. It also strikes me that it won't be too obvious to people how to close the query builder. In addition to being able to toggle it with the pen, maybe ad an &times; to dismiss it? Or a "Done" button?

**Comment by lukebjerring** [comment]
I'll add the collapse behaviour in a separate PR.

**Comment by foolip** [comment]
Trying the steps in https://github.com/web-platform-tests/wpt.fyi/pull/625#issuecomment-427486354 I think there's still a bug, at the end I'm still left at `?label=experimental` and showing experimental runs, not stable. Can you try to repro?

**Comment by lukebjerring** [comment]
Confirmed there's a bug; looks like the collapsed label isn't being updated correctly.

**Comment by lukebjerring** [comment]
PTAL

File: webapp/components/test-runs-query.html [hunk 1]
```diff
@@ -132,7 +132,7 @@
         }
 
         clearQuery() {
-          this.products = Array.from(DEFAULT_PRODUCTS);
+          this.products = DEFAULT_PRODUCTS.map(p => Object.assign({}, p));
```
Comment (foolip) [inline webapp/components/test-runs-query.html]: Aha. Subtle bug there.

File: webapp/components/test-runs-query.html [hunk 2]
```diff
@@ -143,30 +143,28 @@
...
             this.sha = params.sha;
           }
-          if ('label' in params) {
-            this.labels = params.label;
-          }
           if ('product' in params) {
             this.products = params.product.map(p => this.parseProductSpec(p));
           }
+          // Expand a global channel label into the separate products
+          let sharedChannel = (params.label || []).find(l => CHANNELS.has(l));
```
Comment (foolip) [inline webapp/components/test-runs-query.html]: Nit: if you just `let sharedChannel` here you can move `params.label.find(l => CHANNELS.has(l))` into the if body. Minimally clearer, take your pick :)

**Pull request merged**